Week5/MAKEME.md

```diff
@@ -1,124 +1,133 @@
+    let numbers = [1, 2, 3, 4];
+    let newNumbers = [];
+
+    for(let i = 0; i < numbers.length; i++) {
+        if(numbers[i] % 2 !== 0) {
+            newNumbers.push(numbers[i] * 2);
+        }
     }
-    threeFive(10, 15, sayThree, sayFive);
 
-    // Should create an array [10,11,12,13,14,15]
-    // and call sayFive, sayThree, sayThree, sayFive  - please make sure you see why these calls are made before you start coding
+    console.log("The doubled numbers are", newNumbers); // [2, 6]
+
     ```
 
+    Rewrite the above program using `map` and `filter` don't forget to use `=>`
 
-2. Please solve this problem using:
-https://www.freecodecamp.com/challenges/repeat-a-string-repeat-a-string
- 1. A for loop
- 2. A while loop
- 3. A do loop
 
-3. Some practice with objects 
-https://www.freecodecamp.com/challenges/construct-javascript-objects-with-functions
+Underneath you see a very interesting small insight in Maartje's work:
 
-4. Nested loops 
-https://www.freecodecamp.com/challenges/nesting-for-loops
+```js
+let monday = [
+        {
+            name     : 'Write a summary HTML/CSS',
+            duration : 180
+        },
+        {
+            name     : 'Some web development',
+            duration : 120
+        },
+        {
+            name     : 'Fix homework for class10',
+            duration : 20
+        },
+        {
+            name     : 'Talk to a lot of people',
+            duration : 200
+        }
+    ];
+
+let tuesday = [
+        {
+            name     : 'Keep writing summary',
+            duration : 240
+        },
+        {
+            name     : 'Some more web development',
+            duration : 180
+        },
+        {
+            name     : 'Staring out the window',
+            duration  : 10
+        },
+        {
+            name     : 'Talk to a lot of people',
+            duration : 200
+        },
+        {
+            name     : 'Look at application assignments new students',
+            duration : 40
+        }
+    ];
+
+let tasks = monday.concat(tuesday);
+```
 
-5. We did some work with arrays - `var arr = [1,2,3]`
-We can also nest arrays inside arrays like this `var arr2d = [[1,2], [3,4], [5,6]]` (for math people you can think of this as a matrix)
-How would you print all the items of an array with 3 dimensions? 
-How about with K dimensions? 
-What if you didn't know how deep the array was nested? (You don't have to write code for this but think about it)
+2. Write a program that does the following below. Use `map` and `filter`. You will also need a `forEach` or a `for` loop for the 'summing up' part.
 
-6. Here are two functions that look like they do the something similar but they print different results. Please explain what's going on here.
+- Collect two days' worth of tasks.
+- Convert the task durations to hours, instead of minutes.
+- Filter out everything that took two hours or more
```

Comment:
I'm not entirely sure what is asked for. Does filter out mean: remove from the collection or keep in the collection?

Reply:
I think we can further improve on this assignment by introducing a third property for each task: `billable` (boolean). We can ask to filter on billable tasks, rather than on an arbitrary task duration (just to make a bit more realistic).